pulsar/consumer.go

```diff
@@ -165,6 +165,9 @@ type Consumer interface {
 	// Unsubscribe the consumer
 	Unsubscribe() error
 
+	// Closed returns a channel indicating that consumer is closed
+	Closed() <-chan struct{}
+
```

Comment:
Why do we need to add this method, It looks like it has never been called?

Reply:
Yep. Idea is to use it like that:
```
select {
  case msg := <- consumer.Chan();
     // handle msg
  case <-consume.Closed()
     return
}
```

So this method is used by user to check if consumer closed or not. We can not close message channel as it shared across partition conumers, so, this is the way of detecting it

Reply:
Hi @omnilight, I am run into the same problem and thank you for providing a fix.

However, I think the new `Closed()` method is quite hard to use. why not just close the `Chan()` and let `Receive()` return an error?

Reply:
Oh, I see https://github.com/apache/pulsar-client-go/pull/303 mentioned that the `Chan()` can't be closed.

BTW, can you also add a log message when `maxRetry` reached and saying that it will not continue retry?

Reply:
I think, yes, let's add

Reply:
@rueian done